test/gtest/type-builder.cpp

```diff
@@ -498,3 +499,31 @@ TEST_F(EquirecursiveTest, CanonicalizeBasicTypes) {
 TEST_F(IsorecursiveTest, CanonicalizeBasicTypes) {
   testCanonicalizeBasicTypes();
 }
+
+// Test SubTypes utility code.
+TEST_F(NominalTest, testSubTypes) {
+  // Build type types, the second of which is a subtype.
+  TypeBuilder typeBuilder(2);
+  typeBuilder[0] = Struct({Field(Type::anyref, Immutable)});
+  typeBuilder[1] = Struct({Field(Type::funcref, Immutable)});
+  typeBuilder[1].subTypeOf(typeBuilder[0]);
+  auto built = *typeBuilder.build();
+
+  // Build a tiny wasm module that uses the types, so that we can test the
+  // SubTypes utility code.
+  Module wasm;
+  Builder wasmBuilder(wasm);
+  wasm.addFunction(wasmBuilder.makeFunction(
+    "func",
+    Signature(Type::none, Type::none),
+    {Type(built[0], Nullable), Type(built[1], Nullable)},
+    wasmBuilder.makeNop()));
+  SubTypes subTypes(wasm);
+  auto subTypes0 = subTypes.getStrictSubTypes(built[0]);
+  assert(subTypes0.size() == 1 && subTypes0[0] == built[1]);
```

Comment:
Using the gtest macros will ensure that we get nice output and don't stop other tests from running if this test ever starts failing. The `ASSERT` macro variants will return from the test function if they fail and the `EXPECT` macro variants will continue to execute the rest of the function even if they fail, so generally it's best to prefer `EXPECT` except in cases where a failed check means the rest of the test isn't meaningful, like for a bounds check.

```suggestion
  ASSERT_EQ(subTypes0.size(), 1u);
  EXPECT_EQ(subTypes0[0], built[1]);
```

Reply:
Thanks, sounds good. Rewrote to use those.